docs/interface/Metadata.md

```diff
@@ -132,6 +132,14 @@ classes may be allocated. The behaviours are the following:
   `Metadata::WithFluxes` and `Metadata::FillGhosts` to send flux
   corrections across meshblocks.
 
+- If `Metadata::RemeshComm` is set, the variable is communicated between 
+  ranks during remeshing. Variables with `Metadata::Independent` and/or 
+  `Metadata::FillGhost` are also automatically communicated when a block 
+  is communicated from one process to another. Other variables **are not** 
+  communicated across ranks, which has the possibility to cause downstream 
+  codes to produce different results when the same problem is run on 
+  different numbers of ranks. 
```

Comment:
I'm not sure I follow the comment.
How can running on a different number of ranks produce different results?
I could image this happening when a variable is neither `independent` nor `dependent`, but think we should not allow for that case.

Reply:
When a block is moved in the block list within a rank, all of the fields are copied over. When a block is moved between ranks, only independent data is transferred on the assumption that the dependent data can be reconstituted on the new rank from the dependent data. This is normally the case, but sometimes it may be the case that it is too expensive or introduces numerical noise to rebuild some `dependent` fields. Then, it is useful to have a flag that indicates the data in particular dependent variables should be communicated during load balancing.

Reply:
> When a block is moved in the block list within a rank, all of the fields are copied over. 

I thought we'd should move around the pointer to the block and update the data within the block and not copy the data itself.

> When a block is moved between ranks, only independent data is transferred on the assumption that the dependent data can be reconstituted on the new rank from the dependent data.

Yes, this is/was also my working assumption when I wrote the comment.

> This is normally the case, but sometimes it may be the case that it is too expensive or introduces numerical noise to rebuild some dependent fields. Then, it is useful to have a flag that indicates the data in particular dependent variables should be communicated during load balancing.

I'm probably a little hesitant as I'm afraid that the proposed functionality/fix hides a deeper problem and/or invites for some "unfavorable" design choices downstream.
While I understand that there's some extra cost involved in order to rebuild dependent data, I'm currently weighing the "strict" design decision that "everything derived is handled by `FillDerived` at clear points in the dataflow" as a stronger argument than performance (but probably your performance use case is vastly different from the ones I have in mind).

How about we discuss this tomorrow during the call? I'd like to hear feedback from more people on this as I consider this a breaking change (with respect to the underlying design decision).

Reply:
Sure, I am happy to discuss it during the call tomorrow.